core/src/main/java/com/valtech/aemsaas/core/internal/models/sling/search/SearchResultsImpl.java

```diff
@@ -0,0 +1,108 @@
+import com.fasterxml.jackson.annotation.JsonInclude.Include;
+import com.valtech.aemsaas.core.models.search.payload.DefaultFulltextSearchGetRequestPayload;
+import com.valtech.aemsaas.core.models.search.payload.FulltextSearchGetRequestPayload;
+import com.valtech.aemsaas.core.models.search.query.DefaultLanguageQuery;
+import com.valtech.aemsaas.core.models.search.query.DefaultTermQuery;
+import com.valtech.aemsaas.core.models.search.query.HighlightingTagQuery;
+import com.valtech.aemsaas.core.models.search.query.PaginationQuery;
+import com.valtech.aemsaas.core.models.search.results.FulltextSearchResults;
+import com.valtech.aemsaas.core.models.search.results.Result;
+import com.valtech.aemsaas.core.models.sling.search.Search;
+import com.valtech.aemsaas.core.models.sling.search.SearchResults;
+import com.valtech.aemsaas.core.services.search.FulltextSearchConfigurationService;
+import com.valtech.aemsaas.core.services.search.FulltextSearchService;
+import com.valtech.aemsaas.core.utils.StringToInteger;
+import com.valtech.aemsaas.core.utils.request.RequestParameters;
+import com.valtech.aemsaas.core.utils.resource.ParentResource;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import java.util.OptionalInt;
+import javax.annotation.PostConstruct;
+import lombok.Getter;
+import lombok.NonNull;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.sling.api.SlingHttpServletRequest;
+import org.apache.sling.models.annotations.DefaultInjectionStrategy;
+import org.apache.sling.models.annotations.Exporter;
+import org.apache.sling.models.annotations.Model;
+import org.apache.sling.models.annotations.injectorspecific.OSGiService;
+import org.apache.sling.models.annotations.injectorspecific.Self;
+
+@Model(adaptables = SlingHttpServletRequest.class,
+    adapters = {SearchResults.class, ComponentExporter.class},
+    defaultInjectionStrategy = DefaultInjectionStrategy.OPTIONAL,
+    resourceType = RESOURCE_TYPE)
+@Exporter(name = ExporterConstants.SLING_MODEL_EXPORTER_NAME, extensions = ExporterConstants.SLING_MODEL_EXTENSION)
+public class SearchResultsImpl implements SearchResults {
+
+  public static final String RESOURCE_TYPE = "saas-aem-module/components/saas/searchresults";
+
+
+  @Self
+  private SlingHttpServletRequest request;
+
+  @OSGiService
+  private FulltextSearchService fulltextSearchService;
+
+  @OSGiService
+  private FulltextSearchConfigurationService fulltextSearchConfigurationService;
+
+  @JsonInclude(Include.NON_EMPTY)
+  @Getter
+  private String term;
+
+  @JsonInclude(Include.NON_EMPTY)
+  @Getter
+  private int startPage;
+
+  @JsonInclude(Include.NON_EMPTY)
+  @Getter
+  private int resultsPerPage;
+
+  @JsonInclude(Include.NON_NULL)
+  @Getter
+  private List<Result> results;
+
+  @PostConstruct
+  private void init() {
+    RequestParameters requestParametrs = new RequestParameters(request);
+    term = requestParametrs.getParameter("term");
```

Comment:
Please use constant for the strings. 
Also please change name to `q` instead of `term`, just because we would be backwards compatible with dispatcher configs from v1